src/runtime/ValueInlines.h

```diff
@@ -724,14 +724,58 @@ inline double Value::toNumber(ExecutionState& state) const
 
 inline Value Value::toNumeric(ExecutionState& state) const
 {
-    // Let primValue be ? ToPrimitive(value, hint Number).
-    auto primValue = toPrimitive(state);
-    // If Type(primValue) is BigInt, return primValue.
-    if (UNLIKELY(primValue.isBigInt())) {
-        return primValue;
-    }
-    // Return ? ToNumber(primValue).
-    return Value(primValue.toNumber(state));
+// Let primValue be ? ToPrimitive(value, hint Number).
+// If Type(primValue) is BigInt, return primValue.
+// Return ? ToNumber(primValue).
+
+// fast path
+#ifdef ESCARGOT_64
+    auto n = u.asInt64 & TagTypeNumber;
+    if (LIKELY(n)) {
+        return *this;
+    }
+#else
+    if (LIKELY(isInt32())) {
+        return *this;
+    } else if (isDouble()) {
+        return *this;
+    }
+#endif
+    else if (isUndefined()) {
+        return Value(std::numeric_limits<double>::quiet_NaN());
+    } else if (isNull()) {
+        return Value(0);
+    } else if (isBoolean()) {
+        return Value(asBoolean() ? 1 : 0);
+    } else {
+        return toNumericSlowCase(state);
+    }
+}
+
+inline std::pair<Value, bool> Value::toNumericWithTypeInformation(ExecutionState& state) const // <Value, isBigInt>
+{
```

Comment:
Operation of `toNumericWithTypeInformation` is almost same to that of `toNumeric` except that it returns with a boolean which represents if the value is BigInt or not.
I'm wondering that it is worth adding a new function for this.
Or `toNumericWithTypeInformation` could recyle the existing `toNumeric` method in itself with additional boolean return.
What do you think about it?

Reply:
You are right. almost source code are same but there is no way to reduce duplication.
I make toNumericWithTypeInformation for performance. almost operations after toNumberic needs isBigInt or isNumber.
I think we can remove toNumberic function

Reply:
Oops, both functions are used on hot spot in interpreter...

Reply:
It looks much better :)